_test/test_utilities_herbert/test_str_make_cellstr.m

```diff
@@ -0,0 +1,146 @@
+            % Make some 1D character arrays
+            S.ch1 = 'Hello     ';
+            S.ch2 = 'Mister Man';
+            S.ch3 = '  Yellow  ';
+            
+            % Make some cell arrays with a single 1D character array each
+            S.c1 = cellstr(S.ch1);
+            S.c2 = cellstr(S.ch2);
+            S.c3 = cellstr(S.ch3);
+            
+            % Matlab strings
+            S.s1 = string(S.ch1);
+            S.s2 = string(S.ch2);
+            S.s3 = string(S.ch3);
+            
+            % Arrays
+            S.charr = [S.ch1; S.ch2; S.ch3];    % 2D character array
+            S.carr = {S.ch1, S.ch2, S.ch3};     % 1x3 cell array
+            S.sarr = [S.s1, S.s2, S.s3];        % 1x3 string array
+            
+            S.carr_deblank = deblank(S.carr');  % column
+            S.carr_trim = strtrim(S.carr');     % column
+        end
+        
+        %-------------------------------------------------------------------
+        function test_charArray (S)
+            % Tests that a character array is stripped of trailing blanks
+            [ok,cout] = str_make_cellstr(S.ch1);
+            assertTrue(ok)
+            assertEqual(cout, {deblank(S.ch1)})
+        end
+        
+        %-------------------------------------------------------------------
+        function test_multipleCharArrays (S)
+            % Tests that output from multiple character array input is a column
+            % cell array, and stripped of trailing (but not leading) blanks
+            [ok,cout] = str_make_cellstr(S.ch1, S.ch2, S.ch3);
+            assertTrue(ok)
+            assertEqual(cout, S.carr_deblank)
+        end
+        
+        %-------------------------------------------------------------------
+        function test_multipleCharArrays_stringArray_cellstr (S)
+            % Test multiple forms of input, and that string arrays and
+            % cellstr are NOT stripped of trailing blanks, as Matlab behaviour
+            % dictates
+            [ok,cout] = str_make_cellstr([S.ch2; S.ch3], S.sarr, S.carr);
+            assertTrue(ok)
+            assertEqual(cout, [S.carr_deblank(2:3); S.carr'; S.carr'])
+        end
+        
+        %-------------------------------------------------------------------
+        function test_emptyTypes (~)
+            % Test that empty character arrays, cellstr and strings result in
+            % empty character arrays in a cell array (essentially a test of type
+            % conversion)
+            [ok,cout] = str_make_cellstr('', {}, {''}, "");
+            assertTrue(ok)
+            assertEqual(cout, {''; ''; ''})
+        end
+        
+        %-------------------------------------------------------------------
+        function test_char2D (~)
+            % Test 2D character array
+            ch2D = ['Hello     '; 'Mister Man'; '          '; '  Yellow  '];
+            cout_ref = {'Hello'; 'Mister Man'; ''; '  Yellow'};
+            [ok,cout] = str_make_cellstr(ch2D);
```

Comment:
It follows the convention of the Matlab cellstr, which strips trailing space of 2D character arrays when it turns them into a cell array of character vectors - indeed the refactored str_make_cellstr now makes a straightforward call to cellstr.

In answer to your general comment about some of these utilities dressing matlab functions in a way that increases maintenance, that is exactly what has happened. However,  they are used in a number of places within Horace, and I use them in a large number of places in code that I've written for data analysis over the past 15 years. My goal here was to tidy up relatively swiftly a number of problems and inconsistencies. Likewise, while one or two names are misleading, rather than embark on an open-ended task of redefining a string handling utility set with deprecated functions that were aliased to new ones etc., I decided that improving the in-code documentation was more expedient, with one or two additional functions to handle the 'new' (R2017a) Matlab string class. I want to defer an overhaul of this and other code dating back two decades that are ripe for rationalisation (e.g. all those parsing functions) until Horace core has settled down.